db/migrate/20200617183058_create_ahoy_visits_and_events.rb

```diff
@@ -0,0 +1,61 @@
+class CreateAhoyVisitsAndEvents < ActiveRecord::Migration[6.0]
+  def change
+    create_table :ahoy_visits do |t|
+      t.string :visit_token
+      t.string :visitor_token
+
+      # the rest are recommended but optional
+      # simply remove any you don't want
+
+      # user
+      t.references :user
+
+      # standard
+      t.string :ip
```

Comment:
Privacy related question: should we track IPs? We do scramble IPs when we send them to Google Analytics. 

https://github.com/thepracticaldev/dev.to/blob/478b048fc5aa0c5607e8f3491e68811a46f044bf/app/controllers/ga_events_controller.rb#L31-L34

Can we do without them altogether for in house tracking? Or at least encrypt them?

A visit is tied to the user model, which means we're going to have multiple DBs (in each community) that associate user data with their IP.

Reply:
Also, the GDPR considers IP addresses personally identifiable information, so we'd have to remove them for European users on their request.

Reply:
Yeah. We do not currently track or persist IP for any users for max simplicity and compatibility with GDPR, so let's keep it that way.

Reply:
@rhymes @citizen428 @benhalpern Agree with you on all accounts re: privacy concerns. I specifically left all the columns in so we could decide on which we wanted to keep/remove 😉 (I should have left a note about that, sorry!).

I think it sounds like we should just probably remove _all_ of the columns that are not strictly required (basically what @citizen428 mentioned), and then incrementally and intentionally add back whatever we feel is necessary.

Reply:
fwiw  neither of those values in our case, are going to be the actual user IP, those will be Fastly IPs